jupyter_collaboration/handlers.py

```diff
@@ -111,6 +111,7 @@ def exception_logger(exception: Exception, log: Logger) -> bool:
                     file = self._file_loaders[file_id]
                     updates_file_path = f".{file_type}:{file_id}.y"
                     ystore = self._ystore_class(path=updates_file_path, log=self.log)
+                    await ystore.start()
```

Comment:
In the general case, starting a service can be blocking. Actually, after https://github.com/jupyter-server/pycrdt-websocket/pull/42 starting a YStore will be.
```suggestion
                    self.create_task(ystore.start())
                    await ystore.started.wait()
```

Reply:
Actually, starting the YStore should not be done here but [in the DocumentRoom](https://github.com/jupyterlab/jupyter-collaboration/blob/1f367f58a1090ca27e920352b376edf03375b6c7/jupyter_collaboration/rooms.py#L102), since it is optional.
Also, you might want to stop the YStore when stopping the YRoom.

Reply:
I think adding await in the DocumentRoom is a good idea and it also help prevent starting the room at two places which could cause race condition as well. I will make that change.